background/background.js

```diff
@@ -341,4 +351,14 @@ function onRuntimeMessage(request, sender, sendResponse) {
         .catch(() => sendResponse(null));
       return KEEP_CHANNEL_OPEN;
   }
+
+  function responseWithError(pending) {
```

Comment:
> is it really necessary to print the error in the background console?

No. It's a habit to print out unhandled error. Should I remove it?

> it'd be simpler to invoke queryUsercss with sendResponse as a parameter and remove responseWithError altogether. I don't like global auxiliary functions, we'll get rid of them someday.

Both `saveUsercss` and `queryUsercss` share `respondWithError`. If we are going to remove it, they would expand to:
```diff
     case 'saveUsercss':
       // FIXME: both 'saveUsercss' and 'saveStyle' use saveStyle() function,
       // however the response is different so we use a new method here.
-      respondWithError(saveStyle(request));
+      saveStyle(request)
+        .then(r => sendResponse([null, r]))
+        .catch(e => sendResponse([e, null]));
       return KEEP_CHANNEL_OPEN;

     case 'queryUsercss':
-      respondWithError(queryUsercss(request));
+      queryUsercss(request)
+        .then(r => sendResponse([null, r]))
+        .catch(e => sendResponse([e, null]));
       return KEEP_CHANNEL_OPEN;

     case 'healthCheck':

```
It looks cleaner with `respondWithError` to me.

Reply:
1. If an error is caused by external source (like timeout, bad server response, invalid syntax) it should be displayed to the user via standard dialogs like alert. If the error is caused by your uncertainty in the code you write, the cause should be investigated and eliminated. We have one or two `console` from the old times and keep them just because no one investigated their usefulness. AFAIK they are absolutely useless as I never saw them being posted in bug reports.
2. Like I said I don't like global auxiliary functions especially in the background page. It should be possible to rework the code to avoid the need for additional then/catch and respond with a single object that can be sent back to the content script.

Reply:
With the 2nd point, my main objection is that you're *thunking* the function, which is similar to monkeypatching, and it means bad architecture. Of course our current architecture is quite bad, but it doesn't mean we should worsen it.